src/avatar/avatar.component.ts

```diff
@@ -107,26 +109,23 @@ export class IgxAvatar {
     }
 
     private generateInitials(size) {
-        const canvas = document.createElement("canvas");
         const fontSize = size / 2;
-        let ctx;
-
-        canvas.width = size;
-        canvas.height = size;
-
-        ctx = canvas.getContext("2d");
-        ctx.fillStyle = this.bgColor;
-        ctx.fillRect(0, 0, canvas.width, canvas.height);
-        ctx.textAlign = "center";
-        ctx.fillStyle = this.color;
-        ctx.font = fontSize + `px ${this.fontName}`;
-        ctx.fillText(
-            this.initials.toUpperCase(),
-            size / 2,
-            size - size / 2 + fontSize / 3
-        );
-
-        return canvas.toDataURL("image/png");
+
+        const svg = this.initialsImage.nativeElement;
+        svg.setAttribute("width", size);
+        svg.setAttribute("height", size);
+
+        const svgText = svg.children[0];
+        if (svgText) {
+            svgText.textContent = this.initials.toUpperCase();
+            svgText.setAttribute("font-size", fontSize);
+            svgText.setAttribute("font-family", this.fontName);
+
+            const x = fontSize;
+            svgText.setAttribute("x", x);
+            const y = size - size / 2 + fontSize / 3;
+            svgText.setAttribute("y", y);
+        }
```

Comment:
Attributes for width and height should be set with HostBinding decorators to support server side rendering and web workers. Also, don't apply font-size and font-family here, should be done with CSS. If custom font-size and font-family must be specified on component instance level use HostBinding here as well.